File: mobile/lib/features/invoicing/presentation/widgets/_missing_fields_copy.dart
import '../../domain/entities/missing_field.dart';

/// Maps a `MissingField` token returned by the backend to a short
/// French label suitable for the completion modal list.
///
/// Mirrors `web/src/features/invoicing/components/missing-fields-copy.ts`
/// verbatim so users get the same wording across web and mobile.
const Map<String, String> kMissingFieldLabels = {
  'legal_name': 'Raison sociale ou nom légal',
  'trading_name': 'Nom commercial',
  'legal_form': 'Forme juridique',
  'tax_id': 'Numéro SIRET ou identifiant fiscal',
  'vat_number': 'Numéro de TVA intracommunautaire',
  'address_line1': 'Adresse',
  'postal_code': 'Code postal',
  'city': 'Ville',
  'country': 'Pays',
  'invoicing_email': 'Email de facturation',
  'profile_type': 'Type de profil (particulier ou entreprise)',
};

/// Maps the `reason` token to a short qualifier — kept compact so the
/// modal stays scannable. The label is shown before the dash.
const Map<String, String> kMissingFieldReasonLabels = {
  'required': 'obligatoire',
  'invalid_format': 'format invalide',
  'not_validated': 'non validé',
};

/// Returns a French label for the given snake_case field token. Falls
/// back to the underscored token spaced out so a future backend field
/// is still readable until this map is updated.
String fieldLabel(String field) {
  return kMissingFieldLabels[field] ?? field.replaceAll('_', ' ');
}

/// Returns a French label for the given reason token, or null when the
/// token is unknown (so callers can omit the qualifier entirely rather
/// than printing the raw string).
String? reasonLabel(String reason) {
  return kMissingFieldReasonLabels[reason];
}

/// Composes the label rendered as one bullet inside the gate modal:
/// `"Code postal — obligatoire"` when both tokens are known,
/// `"Code postal"` when only the field is known.
String describeMissing(MissingField field) {
  final reason = reasonLabel(field.reason);
  final label = fieldLabel(field.field);
  return reason == null ? label : '$label — $reason';
}

File: web/src/shared/lib/billing-profile/missing-fields-copy.ts
import type { MissingField } from "@/shared/types/billing-profile"

/**
 * Maps a `MissingField` token returned by the backend to a short
 * French label suitable for a list inside the completion modal.
 *
 * The backend ships `field` as a snake_case identifier; the modal
 * renders the label, NEVER the raw token. Unknown tokens fall back
 * to the field name with underscores replaced — so a future
 * field added on the backend is still readable until this map
 * is updated.
 */
const FIELD_LABELS: Record<string, string> = {
  legal_name: "Raison sociale ou nom légal",
  trading_name: "Nom commercial",
  legal_form: "Forme juridique",
  tax_id: "Numéro SIRET ou identifiant fiscal",
  vat_number: "Numéro de TVA intracommunautaire",
  address_line1: "Adresse",
  postal_code: "Code postal",
  city: "Ville",
  country: "Pays",
  invoicing_email: "Email de facturation",
  profile_type: "Type de profil (particulier ou entreprise)",
}

/**
 * Maps the `reason` token to a short qualifier — kept compact so
 * the modal stays scannable. The label is shown before the dash.
 */
const REASON_LABELS: Record<string, string> = {
  required: "obligatoire",
  invalid_format: "format invalide",
  not_validated: "non validé",
}

export function fieldLabel(field: string): string {
  return FIELD_LABELS[field] ?? field.replace(/_/g, " ")
}

export function reasonLabel(reason: string): string {
  return REASON_LABELS[reason] ?? reason.replace(/_/g, " ")
}

export function describeMissing(field: MissingField): string {
  const reason = REASON_LABELS[field.reason] ?? null
  const label = fieldLabel(field.field)
  return reason ? `${label} — ${reason}` : label
}
